### PR TITLE
Update firewalls in maintenance reconciliation

### PR DESCRIPTION
**Pull request by Gerrit91**
This PR implements the set roll for firewalls when a shorthand OS image name is being used and the maintenance reconciliation was triggered.

### CODE REVIEW

File: controllers/deployment/reconcile.go [hunk 1]
```diff
@@ -215,15 +215,13 @@ func sizeHasChanged(newS *v2.FirewallSpec, oldS *v2.FirewallSpec) bool {
 }
 
 func osImageHasChanged(ctx context.Context, m metalgo.Client, newS *v2.FirewallSpec, oldS *v2.FirewallSpec) (bool, error) {
-	if newS.Image != oldS.Image {
-		image, err := m.Image().FindLatestImage(image.NewFindLatestImageParams().WithID(newS.Image).WithContext(ctx), nil)
-		if err != nil {
-			return false, fmt.Errorf("latest firewall image not found:%s %w", newS.Image, err)
-		}
+	image, err := m.Image().FindLatestImage(image.NewFindLatestImageParams().WithID(newS.Image).WithContext(ctx), nil)
```
[the remark this answers is not in the document]
Comment (Gerrit91) [reply controllers/deployment/reconcile.go:218]: Yes, this would be good addition.

**Comment by majst01** [comment]
Deployment does not change and therefor the imagecache will not be asked to fetch the newest image.

It should depend on the gardener `maintain` reconciliation and trigger firewall roll during the maintenance window

**Comment by Gerrit91** [comment]
Looks like this finally works now.

File: controllers/deployment/controller.go [hunk 2]
```diff
@@ -29,6 +35,14 @@ func SetupWithManager(log logr.Logger, recorder record.EventRecorder, mgr ctrl.M
 		log:             log,
 		recorder:        recorder,
 		lastSetCreation: map[string]time.Time{},
+		imageCache: cache.New(5*time.Minute, func(ctx context.Context, id string) (*models.V1ImageResponse, error) {
```
Comment (majst01) [inline controllers/deployment/controller.go:38]: every 5 minute is brutal when we release new versions of images ever 3-4 weeks.

**Review by majst01 — CHANGES_REQUESTED** [review]

**Review by majst01 — APPROVED** [review]

**Pull request merged**